addons/web/models/models.py

```diff
@@ -371,6 +369,21 @@ def formatted_read_group(self, domain, groupby=(), aggregates=(), having=(), off
 
         return self._web_read_group_format(groupby, aggregates, groups)
 
+    def _web_pre_process_aggregates(self, aggregates):
+        # Avoid recordset in _web_read_group_format as aggregate + Add currency_field aggregates for monetary aggregates
+        return tuple(set(
```

Comment:
Try to avoid iterating on `set` (especially in the ORM/web). 
Why ? `set` is Python are not ordered, then you will ended up with an order depending of the hash value of elements. Also, in Python, `hash("string")` depends on an environment variable (`PYTHONHASHSEED`) that are set randomly when your start Python. That makes `set` of string nondeterminism, and that's very painful to debug, to reproduce a bug and may be harmful for caches at some level.

You can either use `OrderedSet()` or `unique()` method instead.